token-metadata/program/tests/update.rs

```diff
@@ -161,4 +171,286 @@ mod update {
+
+        let authority = Keypair::from_bytes(&context.payer.to_bytes()).unwrap();
+
+        // Create rule-set for the transfer
+        let (authorization_rules, auth_data) =
+            create_default_metaplex_rule_set(context, authority.dirty_clone()).await;
+
+        let (new_auth_rules, new_auth_data) =
+            create_default_metaplex_rule_set(context, authority.dirty_clone()).await;
+
+        let update_authority = Keypair::from_bytes(&context.payer.to_bytes()).unwrap();
+
+        let mut da = DigitalAsset::new();
+        da.create_and_mint(
+            context,
+            TokenStandard::ProgrammableNonFungible,
+            Some(authorization_rules),
+            Some(auth_data),
+            1,
+        )
+        .await
+        .unwrap();
+
+        let metadata = da.get_metadata(context).await;
+
+        if let Some(ProgrammableConfig::V1 {
+            rule_set: Some(rule_set),
+        }) = metadata.programmable_config
+        {
+            assert_eq!(rule_set, authorization_rules);
+        } else {
+            panic!("Missing rule set programmable config");
+        }
+
+        let delegate = Keypair::new();
+
+        // Set a delegate
+        da.delegate(
+            &mut context,
+            update_authority.dirty_clone(),
+            delegate.pubkey(),
+            DelegateArgs::TransferV1 {
+                amount: 1,
+                authorization_data: None,
+            },
+        )
+        .await
+        .unwrap();
+
+        // Try to clear the rule set.
+        let mut update_args = UpdateArgs::default();
+        let UpdateArgs::V1 { rule_set, .. } = &mut update_args;
+        // remove the rule set
+        *rule_set = RuleSetToggle::Clear;
+
+        let mut builder = UpdateBuilder::new();
+        builder
+            .authority(update_authority.pubkey())
+            .metadata(da.metadata)
+            .mint(da.mint.pubkey())
+            .token(da.token.unwrap())
+            .authorization_rules(authorization_rules)
+            .payer(update_authority.pubkey());
+
+        if let Some(edition) = da.master_edition {
+            builder.edition(edition);
+        }
+
+        let update_ix = builder.build(update_args).unwrap().instruction();
+
+        let tx = Transaction::new_signed_with_payer(
+            &[update_ix],
+            Some(&update_authority.pubkey()),
+            &[&update_authority],
+            context.last_blockhash,
+        );
+
+        let err = context
+            .banks_client
+            .process_transaction(tx)
+            .await
+            .unwrap_err();
+
+        assert_custom_error!(err, MetadataError::CannotUpdateAssetWithDelegate);
+
+        // Try to update the rule set.
+        let mut update_args = UpdateArgs::default();
+        let UpdateArgs::V1 {
+            rule_set,
+            authorization_data,
+            ..
+        } = &mut update_args;
+        // remove the rule set
```

Comment:
Nit: outdated comment should read update the ruleset

Reply:
Thanks, fixed.